### PR TITLE
cxx_qt::Constructor: Add tests

### PR DESCRIPTION
**Pull request by LeonMatthesKDAB**
Follow up on #566

### CODE REVIEW

**Comment by Be-ing** [comment]
CI passes since rebasing after merging #609

File: crates/cxx-qt-gen/src/generator/rust/constructor.rs [hunk 1]
```diff
@@ -115,14 +110,14 @@ fn generate_arguments_struct(
 fn generate_arguments_initialization(
     struct_name: &Ident,
     instance_name: Ident,
-    argument_list: &Option<Vec<Type>>,
+    argument_list: &[Type],
 ) -> Expr {
     let init_arguments = extract_arguments_from_tuple(argument_list, instance_name);
     println!("init_arguments: {:?}", init_arguments);
```
Comment (ahayzen-kdab) [inline crates/cxx-qt-gen/src/generator/rust/constructor.rs:116]: Can we remove this print line ?

When building it causes `[build] init_arguments: []` to appear etc

Comment (LeonMatthesKDAB) [reply crates/cxx-qt-gen/src/generator/rust/constructor.rs:116]: Oh, yes definitely!

File: crates/.gitignore [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# SPDX-FileCopyrightText: 2023 Klarälvdalens Datakonsult AB, a KDAB Group company <info@kdab.com>
+# SPDX-FileContributor: Leon Matthes <leon.matthes@kdab.com>
+#
+# SPDX-License-Identifier: MIT OR Apache-2.0
+
+.lspconfig.lua
```
Comment (ahayzen-kdab) [inline crates/.gitignore:6]: do we need this in our repository ? seems like something that goes in the user global gitignore ?

Comment (LeonMatthesKDAB) [reply crates/.gitignore]: Huh, I didn't know user ignore files were a thing.
Live and Learn I guess.
Done

**Review by ahayzen-kdab — APPROVED** [review]
LGTM :-)

**Pull request merged**